### PR TITLE
7 experience approval

### PR DESCRIPTION
**Pull request by tom08**

### CODE REVIEW

File: exdb/templates/exdb/experience_approval.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+{% load staticfiles %}
+{% block css %}
+    {{ block.super }}
+    <link rel="stylesheet" type="text/css" href="{% static 'exdb/css/experience_approval.css' %}" />
+{% endblock %}
+{% block content %}
+    <form method="POST">
+        {% csrf_token %}
+        <h1>{% trans "Experience Approval" %}</h1>
+        <a href="{% url 'pending' %}"><p>Return to Pending Experiences</p></a>
```
Comment (enki211) [inline exdb/templates/exdb/experience_approval.html]: Should you wrap the a tag in the p tag instead?

File: exdb/templates/exdb/experience_approval.html [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                <tr>
+                    <th>{% trans "Guest Office" %}:</th>
+                    <td>{{ experience.guest_office }}</td>
+                </tr>
+            {% endif %}
+        </table>
+        <table>
+            <tr><th>{% trans "Comments" %}</th></tr>
+            {% for comment in comments %}
+                <tr><td>{% trans comment.message %}</td></tr>
```
Comment (enki211) [inline exdb/templates/exdb/experience_approval.html]: I don't think we should translate the message

File: exdb/templates/exdb/experience_approval.html [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            {% empty %}
+            <tr><td>{% trans "No comments" %}</td></tr>
```
Comment (enki211) [inline exdb/templates/exdb/experience_approval.html]: This should be indented

File: exdb/views.py [hunk 4]
```diff
@@ -45,3 +47,36 @@ class PendingApprovalQueueView(ListView):
...
+    def get_experience(self):
+        return get_object_or_404(Experience, pk=self.kwargs['pk'], status='pe')
+
+    def get_context_data(self, **kwargs):
+        context = super(ExperienceApprovalView, self).get_context_data()
+        context['experience'] = self.get_experience()
+        context['comments'] = ExperienceComment.objects.filter(experience=context['experience'])
+        return context
+
+    def form_valid(self, form):
```
Comment (enki211) [inline exdb/views.py]: You do not set the approver

File: exdb/views.py [hunk 5]
```diff
@@ -45,3 +47,36 @@ class PendingApprovalQueueView(ListView):
...
+        form.instance.author = self.request.user
+        form.instance.experience = self.get_experience()
+        form.instance.experience.status = 'ad' if self.request.POST.get('approve') else 'de'
+        form.instance.experience.save()
+        return super(ExperienceApprovalView, self).form_valid(form)
+
+    def form_invalid(self, form):
```
Comment (enki211) [inline exdb/views.py]: You do not set the approver

File: exdb/views.py (hunk 4 above)
Comment (aaaustin10) [inline exdb/views.py]: This has been addressed.

File: exdb/templates/exdb/experience_approval.html [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    <link rel="stylesheet" type="text/css" href="{% static 'exdb/css/experience_approval.css' %}" />
+{% endblock %}
+{% block content %}
+    <form method="POST">
+        {% csrf_token %}
+        <h1>{% trans "Experience Approval" %}</h1>
+        <p><a href="{% url 'pending' %}">Return to Pending Experiences</a></p>
+        <span class="form-errors">{{ form.non_field_errors }}</span>
+        <hr/>
+        <table>
```
Comment (shaneallgeier) [inline exdb/templates/exdb/experience_approval.html]: There's no need for this to be a `<table>`.

**Comment by enki211** [comment]
Looks good

**Comment by shaneallgeier** [comment]
I approve.

File: exdb/tests/integration_tests.py [hunk 7]
```diff
@@ -227,6 +227,67 @@ def test_valid_past_experience_creation_view_submit(self):
                          "Experience should have been saved with completed status")
 
 
+class ExperienceApprovalViewTest(StandardTestCase):
+
+    def test_gets_correct_experience(self):
+        e = self.create_experience('pe')
+        self.create_experience('pe')
+        client = Client()
```
Comment (aaaustin10) [inline exdb/tests/integration_tests.py]: You should avoid repeating code wherever possible.

**Pull request merged**